module/User/src/Model/ApiPrincipal.php

```diff
@@ -51,12 +68,53 @@ public function getId(): ?int
 
     public function getToken(): string
     {
+        return str_repeat('*', strlen($this->token) - 5) . substr($this->token, -5);
+    }
+
+    /**
+     * Generate a (new) token and return it
+     * We do not provide a way of specifying a token
+     */
+    public function generateToken(): string
+    {
+        $this->token = Rand::getString(64);
```

Comment:
Please use the PHP native `random_bytes` or `\Random\Randomizer::getBytes()`.

Reply:
b48bbb2e9966b14482d8bd2b22ee043892724aac